### PR TITLE
Feat(compiler-base-session): Add some method for 'Session'.

### PR DESCRIPTION
**Pull request by zong-zhe**
#### 1. Does this PR affect any open issues?(Y/N) and add issue references (e.g. "fix #123", "re #123".):

- [ ] N
- [x] Y 

issue #115

#### 2. What is the scope of this PR (e.g. component or file name):

compiler-base-session

#### 3. Provide a description of the PR(e.g. more details, effects, motivations or doc link):

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [x] Other

add constructor `new_with_src_code()` to 'Session' for constructing by source code.
add method 'emit_err()' to 'Session' for displaying error diagnostic.

#### 4. Are there any breaking changes?(Y/N) and describe the breaking changes(e.g. more details, motivations or doc link):

- [x] N
- [ ] Y 

#### 5. Are there test cases for these changes?(Y/N) select and add more details, references or doc links:

- [x] Unit test
- [ ] Integration test
- [ ] Benchmark (add benchmark stats below)
- [ ] Manual test (add detailed scripts or steps below)
- [ ] Other

#### 6. Release note

Please refer to [Release Notes Language Style Guide](https://kusionstack.io/docs/governance/release-policy/) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: compiler_base/session/src/lib.rs [hunk 1]
```diff
@@ -26,24 +29,99 @@ impl Session {
...
+    ///    sess.emit_err(MyError {}).unwrap();
+    /// });
+    /// assert!(result.is_err());
+    ///
+    /// ```
+    pub fn emit_err(&self, err: impl SessionDiagnostic) -> Result<bool> {
+        let diag = err
+            .into_diagnostic(self)
+            .with_context(|| "Internale Bug: Fail to display error diagnostic")?;
+        self.diag_handler
```
Comment (Peefy) [inline compiler_base/session/src/lib.rs:117]: Just like `main.rs`, U can use `self.diag_handler.add_err_diagnostic(diag)?.abort_if_errors()?` to simplify it.

File: compiler_base/session/src/lib.rs [hunk 2]
```diff
@@ -26,24 +29,99 @@ impl Session {
...
+    ///    // 4. Emit the error diagnostic.
+    ///    sess.emit_err(MyError {}).unwrap();
+    /// });
+    /// assert!(result.is_err());
+    ///
+    /// ```
+    pub fn emit_err(&self, err: impl SessionDiagnostic) -> Result<bool> {
+        let diag = err
+            .into_diagnostic(self)
+            .with_context(|| "Internale Bug: Fail to display error diagnostic")?;
```
Comment (Peefy) [inline compiler_base/session/src/lib.rs:116]: err.into_diagnostic(self)?

File: compiler_base/session/src/lib.rs (hunk 1 above)
Comment (zong-zhe) [reply compiler_base/session/src/lib.rs]: fixed

**Review by Peefy — APPROVED** [review]
LGTM

File: compiler_base/session/src/lib.rs (hunk 2 above)
Comment (zong-zhe) [reply compiler_base/session/src/lib.rs]: fixed

**Review by ldxdl — APPROVED** [review]
LGTM

**Pull request merged**